inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/InstrumentationScopeResolver.java

```diff
@@ -0,0 +1,257 @@
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+import static rocks.inspectit.oce.core.config.model.instrumentation.scope.MethodMatcherSettings.AccessModifier;
+
+/**
+ * This class is used to resolve the {@link InstrumentationScope} based on the {@link InstrumentationScopeSettings} contained
+ * in the configuration.
+ */
+@Component
+@Slf4j
+public class InstrumentationScopeResolver {
+
+    /**
+     * Creates the {@link InstrumentationScope} instances based on the {@link InstrumentationScopeSettings} contained in
+     * the given {@link InstrumentationSettings}.
+     *
+     * @param source the configuration used as basis for the {@link InstrumentationScope}s
+     * @return Returns a map containing {@link InstrumentationScope} instances. The keys are representing a unique identifier for each scope.
+     */
+    public Map<String, InstrumentationScope> resolve(InstrumentationSettings source) {
+        Map<String, InstrumentationScope> scopeMap = new HashMap<>();
+
+        if (source != null && source.getScopes() != null) {
+            for (Map.Entry<String, InstrumentationScopeSettings> scopeEntry : source.getScopes().entrySet()) {
+                InstrumentationScopeSettings scopeSettings = scopeEntry.getValue();
+                if (scopeSettings.getTypeScope() == null) {
+                    log.warn("Processing of scope '{}' skipped. Empty type-scope is not allowed!", scopeEntry.getKey());
+                    continue;
+                }
+
+                log.debug("Processing scope '{}'.", scopeEntry.getKey());
+
+                InstrumentationScope scope = resolveScope(scopeEntry.getKey(), scopeSettings);
+
+                log.debug("|> Type scope: {}", scope.getTypeMatcher() != null ? scope.getTypeMatcher().toString() : "-");
+                log.debug("|> Method scope: {}", scope.getMethodMatcher() != null ? scope.getMethodMatcher().toString() : "-");
+
+                scopeMap.put(scopeEntry.getKey(), scope);
+            }
+        }
+
+        return scopeMap;
+    }
+
+    /**
+     * Resolving the {@link InstrumentationScope} for the given parameters.
+     *
+     * @return The resolve {@link InstrumentationScope}.
+     */
+    private InstrumentationScope resolveScope(String key, InstrumentationScopeSettings scopeSettings) {
+        ElementMatcher.Junction<TypeDescription> typeMatcher = buildTypeMatcher(scopeSettings);
+        ElementMatcher.Junction<MethodDescription> methodMatcher = buildMethodMatcher(scopeSettings, typeMatcher);
+
+        if (typeMatcher == null) {
+            typeMatcher = any();
+            log.warn("|> The defined scope '{}' is not narrowing the type-scope, thus, matching ANY types! This may cause a negative performance impact!", key);
+        }
+
+        return new InstrumentationScope(typeMatcher, methodMatcher);

```

Comment:
shouldn't you use typeMatcher.and(declaresMethod(methodMatcher)) as a better Type Matcher?